wporg-gp-translation-events.php

```diff
@@ -162,12 +162,58 @@ public function event_meta_boxes() {
 	 */
 	public function event_dates_meta_box( WP_Post $post ) {
 		wp_nonce_field( 'event_dates_nonce', 'event_dates_nonce' );
-		$event_start = get_post_meta( $post->ID, '_event_start', true );
-		$event_end   = get_post_meta( $post->ID, '_event_end', true );
-		echo '<label for="event_start">Start Date: </label>';
-		echo '<input type="date" id="event_start" name="event_start" value="' . esc_attr( $event_start ) . '" required>';
-		echo '<label for="event_end">End Date: </label>';
-		echo '<input type="date" id="event_end" name="event_end" value="' . esc_attr( $event_end ) . '" required>';
+		$event_start    = get_post_meta( $post->ID, '_event_start', true );
+		$event_end      = get_post_meta( $post->ID, '_event_end', true );
+		$event_timezone = get_post_meta( $post->ID, '_event_timezone', true );
+		?>
+		<label for="event_start">Start Date (UTC): </label>
+		<input type="datetime-local" id="event_start" name="event_start" value="<?php echo esc_attr( $event_start ); ?>" required><br>
+		<label for="event_end">End Date (UTC): </label>
+		<input type="datetime-local" id="event_end" name="event_end" value="<?php echo esc_attr( $event_end ); ?>" required><br>
+		<label for="event-timezone">Timezone: </label>
+		<select id="event-timezone" name="event_timezone" required>
+			<?php
+			echo wp_kses(
+				wp_timezone_choice( $event_timezone, get_user_locale() ),
+				array(
+					'optgroup' => array( 'label' => array() ),
+					'option'   => array(
+						'value'    => array(),
+						'selected' => array(),
+					),
+				)
+			);
+			?>
+		</select>
+		<?php
+	}
+
+	/**
+	 * Output the event dates meta box.
+	 *
+	 * @param  WP_Post $post The current post object.
+	 */
+	public function hosts_meta_box( WP_Post $post ) {
+		$hosts = explode( ',', get_post_meta( $post->ID, '_hosts', true ) );
```

Comment:
Similarly to my other comment about using the event repository, it think it would make sense to use the `Attendee_Repository` to retrieve the list of hosts:

```php
$hosts = self::get_attendee_repository()->get_hosts( $post->ID )`;
```

Reply:
Implemented, thanks for the code snippet!